src/rnpkeys/rnpkeys.c

```diff
@@ -355,7 +355,10 @@ parse_option(rnp_cfg_t *cfg, int *cmd, const char *s)
 
     if (!compiled) {
         compiled = 1;
-        (void) regcomp(&opt, "([^=]{1,128})(=(.*))?", REG_EXTENDED);
+        if (regcomp(&opt, "([^=]{1,128})(=(.*))?", REG_EXTENDED) != 0) {
+            fprintf(stderr, "Can't compile regex\n");
```

Comment:
@flowher 

> Memory sometimes is allocated even if regcomp fails.

`regfree` should only be called if `regcomp` returns 0, otherwise everything is just on the stack..